From 45a550304ec1888c4d4b88910e54f0f091a6c169 Mon Sep 17 00:00:00 2001
From: "Charles (Chas) Williams" <ciwillia@vyatta.att-mail.com>
Date: Wed, 19 Feb 2020 18:20:09 -0500
Subject: [PATCH] ixgbe: stop delayed interrupt handler when starting

The delayed interrupt handled may not have happen yet when the
adapter is stopped. Cancel this alarm to prevent concurrent
access when restarting.

Fixes: af75078fece3("first public release")

Signed-off-by: Chas Williams <chas3@att.com>
---
 drivers/net/ixgbe/ixgbe_ethdev.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ixgbe/ixgbe_ethdev.c
+++ b/drivers/net/ixgbe/ixgbe_ethdev.c
@@ -2600,6 +2600,9 @@ ixgbe_dev_start(struct rte_eth_dev *dev)
 	/* Stop the link setup handler before resetting the HW. */
 	ixgbe_dev_wait_setup_link_complete(dev, 0);
 
+	/* Stop the delayed interrupt handler. */
+	rte_eal_alarm_cancel(ixgbe_dev_interrupt_delayed_handler, dev);
+
 	/* disable uio/vfio intr/eventfd mapping */
 	rte_intr_disable(intr_handle);
 
